mkdocs/docs/backends/cpu-dylib.md

```diff
@@ -0,0 +1,141 @@
+
+<!-- TODO(??): when to use CPU vs GPU vs other backends -->
+
+## Get runtime and compiler
+
+### Get IREE runtime with dylib HAL driver
+
+You will need to get an IREE runtime that supports the dylib HAL driver
+so it can execute the model on CPU via dynamic libraries containing native
+CPU instructions.
+
+<!-- TODO(??): vcpkg -->
+
+
+#### Build runtime from source
+
+Please make sure you have followed the [Getting started][get-started] page
+to build IREE for Linux/Windows and the [Android cross-compilation][android-cc]
+page for Android. The dylib HAL driver is compiled in by default on all
+platforms.
+
+<!-- TODO(??): a way to verify dylib is compiled in and supported -->
+
+If you want to explicitly specify HAL drivers to support, you will need to add
+`DyLib` to the `IREE_HAL_DRIVERS_TO_BUILD` CMake list variable when configuring
+(for target).
+
+### Get compiler for CPU native instructions
+
+<!-- TODO(??): vcpkg -->
+
+#### Download as Python package
+
+IREE uses continuous integration services to compile and publish artifacts into
+[GitHub Releases][iree-releases] page. Right now these are just snapshots of
+the `main` development branch.
+
+You can install the Python package containing the SPIR-V compiler by
+
+``` shell
+python -m pip install iree-compiler-snapshot \
+    -f https://github.com/google/iree/releases
+```
+
+!!! tip
+    `iree-translate` is installed as `/path/to/python/site-packages/iree/tools/core/iree-translate`.
+    You can find out the full path to the `site-packages` directory via the
+    `python -m site` command.
+
+#### Build compiler from source
+
+Please make sure you have followed the [Getting started][get-started] page
+to build IREE for Linux/Windows and the [Android cross-compilation][android-cc]
+page for Android. The dylib compiler backend is compiled in by default on all
+platforms.
+
+If you want to explicitly specify HAL drivers to support, you will need to add
+`DYLIB-LLVM-AOT` to the `IREE_TARGET_BACKENDS_TO_BUILD` CMake list variable when
+configuring (for host).
+
+## Compile and run the model
+
+With the compiler and runtime for dynamic libraries, we can now compile a model
+and run it on the CPU.
+
+### Compile the model
+
+IREE compilers transform a model into its final deployable format in many
+sequential steps. A model authored with Python in an ML framework should use the
+corresponding framework's import tool to convert into a format (i.e.,
+[MLIR][mlir]) expected by main IREE compilers first.
+
+Using MobileNet v2 as an example, you can download the SavedModel with trained
+weights from [TensorFlow Hub][tf-hub-mobilenetv2] and convert it using IREE's
+[TensorFlow importer][tf-import]. Then,
+
+#### Compile using the command-line
+
+In the build directory, run the following command:
+
+``` shell hl_lines="3"
+iree/tools/iree-translate \
+    -iree-mlir-to-vm-bytecode-module \
+    -iree-hal-target-backends=dylib-llvm-aot \
+    iree_input.mlir -o mobilenet.vmfb
```

Comment:
could differentiate between outputs with some filename tweaks

this doc: `-o mobilenet_dylib.vmfb`
gpu doc: `-o mobilenet_vulkan.vmfb`

(then also update the `iree-run-module` commands to use the new names)